sdk/storage/azure-storage-blob-cryptography/src/test/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClientBuilderTests.java

```diff
@@ -279,6 +281,16 @@ public void getNonEncodedBlobName(String originalBlobName) {
         assertTrue(encryptedBlobClient.getBlobUrl().contains(encodedName));
     }
 
+    @ParameterizedTest
+    @ValueSource(longs = { 0, -1, 15, 4L * Constants.GB })
+    public void illegalRegionLength(long regionLength) {
+        // should we have a test for checking whether length > 4MB?
+        assertThrows(IllegalArgumentException.class, () ->
+            new EncryptedBlobClientBuilder(EncryptionVersion.V2)
+                .blobEncryptionOptions(new BlobEncryptionOptions().setAuthenticatedRegionDataLength(regionLength))
```

Comment:
For this test, we should just be able to instantiate just the encryption options class. We don't necessarily need to create an encrypted blob client to exercise the validation logic.